src/scripts/main.js

```diff
@@ -1,3 +1,72 @@
 'use strict';
 
-// write code here
+const container = document.querySelector('.container');
+const tbody = document.querySelector('tbody');
+let rows = [...tbody.querySelectorAll('tr')];
+
+const appendRowBtn = document.querySelector('.append-row');
+const removeRowBtn = document.querySelector('.remove-row');
+const appendColumnBtn = document.querySelector('.append-column');
+const removeColumnBtn = document.querySelector('.remove-column');
+
+const updateRows = () => {
+  return [...tbody.querySelectorAll('tr')];
+};
+
+const setMaxValue = (list, appendBtn, removeBtn) => {
+  if (list.length < 10) {
+    appendBtn.disabled = false;
+  } else {
+    appendBtn.disabled = true;
+  }
+
+  if (list.length === 2) {
+    removeBtn.disabled = true;
+  } else {
+    removeBtn.disabled = false;
+  }
+};
+
+container.addEventListener('click', e => {
+  const newRow = tbody.children[0].cloneNode(true);
+
+  if (e.target.tagName !== 'BUTTON') {
+    return;
+  }
+
+  switch (e.target.className) {
+    case 'append-row button':
+      tbody.append(newRow);
+      rows = updateRows();
+      break;
+
+    case 'remove-row button':
+      tbody.deleteRow(rows.length - 1);
+      rows = updateRows();
+      break;
+
+    case 'append-column button':
+      rows.map(({ children }) => {
```

Comment:
it's better to use `forEach` here because you're not returning any value
```suggestion
      rows.forEach(({ children }) => {
```